### DISCUSSION TITLE
set atom initial value to result of react hook

### DISCUSSION THREAD

**Discussion by MadadiSaeed** [pmndrs/jotai#1895] [Q&A] [answered]
I am using `useFloating` hook from floating-ui library for react and this hook returns an object. Since i need this result object in child components, i use react context api. So the question is how to create an atom and set it's initial value to this result object.  I need to add this result object as initial value of the atom.

**Comment by dai-shi** [comment, marked as answer]
Atoms are designed to be defined outside React, so initializing with props is not very easy in general.

There are few options:
- `useHydrateAtoms` if it's only initialization
- `useEffect` (maybe with `useHydratedAtoms`) to sync value between props and atoms
- create atom in render, which is possible but tricky, and may require atoms-in-atom pattern
- use `jotai-molecules` for more complex scenario

**Comment by joeyfigaro** [comment]
For anyone finding their way here as of Sep. 2025, `jotai-molecules` is now named [Bunshi](https://jotai.org/docs/extensions/scope#bunshi-formerly-jotai-molecules). ❤️
